plugins/scaffolder-backend/api-report.md

```diff
@@ -211,6 +211,50 @@ export function createGithubActionsDispatchAction(options: {
   JsonObject
 >;
 
+// @public
+export function createGithubDeployKeyAction(options: {
+  integrations: ScmIntegrationRegistry;
+}): TemplateAction_2<
+  {
+    repoUrl: string;
+    publicKey: string;
+    privateKey: string;
+    deployKeyName: string;
+    privateKeySecretName?: string | undefined;
+    token?: string | undefined;
+  },
+  JsonObject
+>;
+
+// @public
+export function createGithubEnvironmentAction(options: {
+  integrations: ScmIntegrationRegistry;
+}): TemplateAction_2<
+  {
+    repoUrl: string;
+    name: string;
+    deploymentBranchPolicy?:
+      | {
+          protected_branches: boolean;
+          custom_branch_policies: boolean;
```

Comment:
Just one final question on this setup. These bools are required. Do they have to be? I'm just thinking that maybe it's hard to add more fields to this without breaking everybody's templates, if they keep being required.

Reply:
The field `deploymentBranchPolicy` is optional but if you specify it, you have to specify both variables, and they're mutually exclusive.  Kind of a kludgy API IMHO but just reflecting how Gitthub's API exposes it.

Maybe I'm missing your point tho...

Reply:
No if it mirrors the actual underlying API then it is what it is, thanks for reporting back!